api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2897,4 +2927,7 @@ service AuthService {
 
   // UpdateClusterMaintenanceConfig updates the current maintenance window config singleton.
   rpc UpdateClusterMaintenanceConfig(types.ClusterMaintenanceConfigV1) returns (google.protobuf.Empty);
+
+  // GetResourceUsage returns the usage data for resources which are limited on usage-based billing plans.
+  rpc GetResourceUsage(GetResourceUsageRequest) returns (GetResourceUsageResponse);
```

Comment:
I addressed more or less all the smaller comments. Let me know what the consensus is on moving this to a dedicated service. Naming is also an open question, since "usage" conflicts somewhat with [usagereporter](https://github.com/gravitational/teleport.e/tree/a2b5f0fe38d19a5c3c93cea2f5cbd14c72d8ffc6/lib/cloud/usagereporter) and whatnot (posthog stuff).

Reply:
> I also think we should create the following PR and move the current device usage API and depreciate the current one.

I'm happy to take that on when the time comes.

Reply:
(Resolving as I already have a TODO in my personal notes to track it.)